src/e3/electrolyt/plan.py

```diff
@@ -21,6 +23,57 @@ class PlanError(E3Error):
     pass
 
 
+class ConditionalConstant:
+    """Allow defining constant in the plan that can depend on external values.
+
+    Note that ConditionalConstant are meant to enable/disable part of the
+    plan but not to modify the plan completely. In particular it is not
+    recommended to write things such as:
+
+        if my_const:
+            anod_build("my_component")
+
+    In that case tools parsing the plan won't be able to detect potential issues.
+    Instead use:
+
+        with defaults(enabled="my_const"):
+             anod_build("my_component")
+    """
+
+    RECORD: List[ConditionalConstant] = []
+
+    def __init__(self, name: str, value: bool) -> None:
+        """Set constant value.
+
+        :param name: constant name for debug info
+        :param value: boolean value
+        """
+        self.value = value
+        self.name = name
+        self.RECORD.append(self)
+
+    def __bool__(self) -> bool:
+        return self.value
+
+    @classmethod
+    def shuffle(cls) -> Iterator[List[ConditionalConstant]]:
+        """Generate all possible set of values for all conditional constants.
+
+        In some cases we want to evaluate the plan for different set of values
+        for the conditional constants to verify that the plan will still be
+        valid under different conditions.
+        """
+        initial_values = tuple(bool(c) for c in cls.RECORD)
+        for shuffeld_values in itertools.product(
+            [True, False], repeat=len(initial_values)
+        ):
+            if shuffeld_values != initial_values:
```

Comment:
FWIW, it looks like there is a typo -> "shuffled"

Reply:
indeed! I'll fix it